core/app/c/[communitySlug]/pubs/[pubId]/components/PubValues.tsx

```diff
@@ -0,0 +1,145 @@
+"use client";
+
+import type { ReactNode } from "react";
+
+import { useState } from "react";
+import Link from "next/link";
+import { Value } from "@sinclair/typebox/value";
+import { getJsonSchemaByCoreSchemaType } from "schemas";
+
+import type { JsonValue } from "contracts";
+import { CoreSchemaType } from "db/public";
+import { Button } from "ui/button";
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "ui/collapsible";
+import { ChevronDown, ChevronRight } from "ui/icon";
+
+import type { FileUpload } from "~/lib/fields/fileUpload";
+import type { FullProcessedPub } from "~/lib/server/pub";
+import { FileUploadPreview } from "~/app/components/forms/FileUpload";
+import { getPubTitle, valuesWithoutTitle } from "~/lib/pubs";
+
+const PubValueHeading = ({
+	depth,
+	children,
+	...props
+}: React.HTMLAttributes<HTMLHeadingElement> & { depth: number }) => {
+	// Pub depth starts at 1
+	switch (depth - 1) {
+		case 0:
+			return <h2 {...props}>{children}</h2>;
+		case 1:
+			return <h3 {...props}>{children}</h3>;
+		case 2:
+			return <h4 {...props}>{children}</h4>;
+		default:
+			return <h5 {...props}>{children}</h5>;
+	}
+};
+
+export const PubValues = ({ pub }: { pub: FullProcessedPub }): ReactNode => {
+	const { values, depth } = pub;
+	if (!values.length) {
+		return null;
+	}
+
+	const filteredValues = valuesWithoutTitle(pub);
+
+	// Group values by field so we only render one heading for relationship values that have multiple entries
+	const groupedValues: Record<string, FullProcessedPub["values"]> = {};
+	filteredValues.forEach((value) => {
+		if (groupedValues[value.fieldName]) {
+			groupedValues[value.fieldName].push(value);
+		} else {
+			groupedValues[value.fieldName] = [value];
+		}
+	});
+	return Object.entries(groupedValues).map(([fieldName, fieldValues]) => {
+		return (
+			<div className="my-2" key={fieldName}>
+				<PubValueHeading depth={depth} className={"mb-2 text-base font-semibold"}>
+					{fieldName}
+				</PubValueHeading>
+				<div className={"ml-2"} data-testid={`${fieldName}-value`}>
+					{fieldValues.map((value) => (
+						<PubValue value={value} key={value.id} />
+					))}
+				</div>
+			</div>
+		);
+	});
+};
+
+const PubValue = ({ value }: { value: FullProcessedPub["values"][number] }) => {
+	const [isOpen, setIsOpen] = useState(false);
+	if (value.relatedPub) {
+		const { relatedPub, ...justValue } = value;
+		const justValueElement = justValue.value ? (
+			<span className="mr-2 italic">{<PubValue value={justValue} />}:</span>
+		) : null;
+		if (relatedPub.isCycle) {
```

Comment:
confusingly the root pub is being returned with `isCycle: true`, which we should investigate. it doesn't cause problems here since we don't check `isCycle` on the root pub, only the related pubs.